share/spack/spack-completion.bash

```diff
@@ -288,7 +288,7 @@ function _spack_debug {
     fi
 }
 
-function _spack_create-db-tarball {
+function _spack_create_db_tarball {
```

Comment:
Ah, I know what the problem is. When invoked as `sh`, `bash` will run in compatibility mode. In POSIX, the only valid characters in function names are underscores and alphanumeric characters. So I think what you'll need to do is change it to `_spack_debug_create_db_tarball` and modify the `subfunction` variable to replace dashes with underscores.